Sources/Sentry/SentrySerialization.m

```diff
@@ -100,32 +101,33 @@ + (SentryEnvelope *_Nullable)envelopeWithData:(NSData *)data
                                                                                error:&error];
             if (nil != error) {
                 SENTRY_LOG_ERROR(@"Failed to parse envelope header %@", error);
-            } else {
-                SentryId *eventId = nil;
-                NSString *eventIdAsString = headerDictionary[@"event_id"];
-                if (nil != eventIdAsString) {
-                    eventId = [[SentryId alloc] initWithUUIDString:eventIdAsString];
-                }
-
-                SentrySdkInfo *sdkInfo = nil;
-                if (nil != headerDictionary[@"sdk"]) {
-                    sdkInfo = [[SentrySdkInfo alloc] initWithDict:headerDictionary];
-                }
-
-                SentryTraceContext *traceContext = nil;
-                if (nil != headerDictionary[@"trace"]) {
-                    traceContext =
-                        [[SentryTraceContext alloc] initWithDict:headerDictionary[@"trace"]];
-                }
-
-                envelopeHeader = [[SentryEnvelopeHeader alloc] initWithId:eventId
-                                                                  sdkInfo:sdkInfo
-                                                             traceContext:traceContext];
-
-                if (headerDictionary[@"sent_at"] != nil) {
-                    envelopeHeader.sentAt = sentry_fromIso8601String(headerDictionary[@"sent_at"]);
-                }
+                break;
+            }
+
+            SentryId *eventId = nil;
+            NSString *eventIdAsString = headerDictionary[@"event_id"];
+            if (nil != eventIdAsString) {
+                eventId = [[SentryId alloc] initWithUUIDString:eventIdAsString];
+            }
+
+            SentrySdkInfo *sdkInfo = nil;
+            if (nil != headerDictionary[@"sdk"]) {
+                sdkInfo = [[SentrySdkInfo alloc] initWithDict:headerDictionary];
             }
+
+            SentryTraceContext *traceContext = nil;
+            if (nil != headerDictionary[@"trace"]) {
+                traceContext = [[SentryTraceContext alloc] initWithDict:headerDictionary[@"trace"]];
+            }
+
+            envelopeHeader = [[SentryEnvelopeHeader alloc] initWithId:eventId
+                                                              sdkInfo:sdkInfo
+                                                         traceContext:traceContext];
+
+            if (headerDictionary[@"sent_at"] != nil) {
+                envelopeHeader.sentAt = sentry_fromIso8601String(headerDictionary[@"sent_at"]);
+            }
+
             break;
```

Comment:
l: What is this break at the end of the loop?

Reply:
It's not at the end of the loop. It's at the end of the if statement. It also confused me a bit when I looked at it the first time 😁

Reply:
But after the If the loops ends, what I'm missing here?

Reply:
Forget about it. I got it now. I was thinking about `continue`